src/main/java/backend/Logic.java

```diff
@@ -200,18 +198,51 @@ public Model getRepo(String repoId) {
      * @return true if label replacement on GitHub was a success, false otherwise.
      */
     public CompletableFuture<Boolean> replaceIssueLabels(TurboIssue issue, List<String> newLabels) {
+        List<String> originalLabels = issue.getLabels();
+
         logger.info("Changing labels for " + issue + " on UI");
-        issue.setLabels(newLabels);
+        Optional<TurboIssue> replaceResult = models.replaceIssueLabels(issue.getRepoId(), issue.getId(), newLabels);
+        if (!replaceResult.isPresent()) {
+            return CompletableFuture.completedFuture(false);
+        }
```

Comment:
Seems like a the chained Futures still run even if the previous Future has completed before the chaining. Did a simple experiment using this:
```java
CompletableFuture<Integer> future = CompletableFuture.completedFuture(1);
future.thenApply((x) -> {
    System.out.println(x);
    return x;
});
```